client/dashboard/domains/domain-transfer/select-site.tsx

```diff
@@ -0,0 +1,153 @@
+import { sitesQuery } from '@automattic/api-queries';
+import { useQuery } from '@tanstack/react-query';
+import {
+	SearchControl,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { DataViews } from '@wordpress/dataviews';
+import { __ } from '@wordpress/i18n';
+import { useMemo, useState, useCallback, useRef } from 'react';
+import { canManageSite } from '../../sites/features';
+import SiteIcon from '../../sites/site-icon';
+import type { Site } from '@automattic/api-core';
+import type { View } from '@wordpress/dataviews';
+import './select-site.scss';
+
+interface Props {
+	attachedSiteId?: number;
+	onSiteSelect: ( site: Site ) => void;
+}
+
+export function SelectSite( { attachedSiteId, onSiteSelect }: Props ) {
+	const { data: allSites = [], isLoading } = useQuery( sitesQuery() );
+	const sites = useMemo(
+		() => allSites.filter( ( site ) => canManageSite( site ) && site.ID !== attachedSiteId ),
+		[ allSites, attachedSiteId ]
+	);
+	const searchInputRef = useRef< HTMLInputElement >( null );
+
+	const perPage = 6;
+	const [ view, setView ] = useState< View >( {
+		type: 'list',
+		perPage,
+		page: 1,
+		search: '',
+		titleField: 'name',
+		descriptionField: 'URL',
+		mediaField: 'icon',
+		infiniteScrollEnabled: true,
+	} );
+
+	const [ selection, setSelection ] = useState< string[] >( [] );
+	const totalPages = Math.ceil( sites.length / perPage );
+	const currentPage = view.page || 1;
+
+	// Handle selection changes
+	const handleSelectionChange = useCallback(
+		( newSelection: string[] ) => {
+			setSelection( newSelection );
+
+			// Get selected site object
+			const selectedSite = sites.find( ( site ) =>
+				newSelection.includes( site.ID?.toString() ?? '' )
+			);
+
+			selectedSite && onSiteSelect( selectedSite );
+		},
+		[ sites, onSiteSelect ]
+	);
+
+	const fields = [
+		{
+			id: 'icon',
+			render: ( { item }: { item: Site } ) => <SiteIcon site={ item } size={ 32 } />,
+		},
+		{
+			id: 'name',
+			label: 'Site Name',
+			render: ( { item }: { item: Site } ) => item.name,
+		},
+		{
+			id: 'URL',
+			label: 'URL',
+			render: ( { item }: { item: Site } ) => {
+				return <Text className="domain-transfer-select-site__url-field">{ item.URL }</Text>;
```

Comment:
Do we want to show the scheme (e.g. `https://`) here? If not, we can use `getSiteDisplayUrl()`, like what we're doing in the Site List.